### PR TITLE
ddl: split range group for each index separately for global sort import

### LINKED ISSUE #50657: Prevent split kv ranges on too much index data when using multi-schema change
## Enhancement

There may be too many `multiFileStat`. It is easy to reach the file limit supported by global sort, especially in scenarios with multi-schema changes.

https://github.com/pingcap/tidb/blob/9f5d2160b108efb953327a5ccad09c3b160b7403/pkg/ddl/backfilling_dist_scheduler.go#L356-L357

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #50657

Problem Summary: 

There may be too many `multiFileStat`. It is easy to reach the file limit supported by global sort, especially in scenarios with multi-schema changes.

https://github.com/pingcap/tidb/blob/9f5d2160b108efb953327a5ccad09c3b160b7403/pkg/ddl/backfilling_dist_scheduler.go#L356-L357

### What changed and how does it work?

When using multi-schema change to add multiple indexes, we split the range group for each index separately.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/backfilling_dist_executor.go [hunk 1]
```diff
@@ -53,10 +53,15 @@ type BackfillTaskMeta struct {
...
-	external.SortedKVMeta `json:",inline"`
+	// Used by read index step.
+	RowStart []byte `json:"row_start"`
+	RowEnd   []byte `json:"row_end"`
+
+	// Used by global sort write & ingest step.
+	RangeSplitKeys [][]byte                 `json:"range_split_keys,omitempty"`
+	DataFiles      []string                 `json:"data_files,omitempty"`
+	StatFiles      []string                 `json:"stat_files,omitempty"`
+	MetaGroups     []*external.SortedKVMeta `json:"meta_groups,omitempty"`
```
Comment (ywqzzy) [inline pkg/ddl/backfilling_dist_executor.go:64]: Add more comments for this field

File: pkg/ddl/backfilling_dist_scheduler.go [hunk 2]
```diff
@@ -426,22 +443,26 @@ func generateMergePlan(
...
+		}
 	}
 	if skipMergeSort(multiStats) {
 		logger.Info("skip merge sort")
 		return nil, nil
 	}
 
 	// generate merge sort plan.
-	_, _, _, multiFileStat, err := getSummaryFromLastStep(taskHandle, task.ID, proto.BackfillStepReadIndex)
+	kvMetaGroups, err := getSummaryFromLastStep(taskHandle, task.ID, proto.BackfillStepReadIndex, logger)
```
Comment (ywqzzy) [inline pkg/ddl/backfilling_dist_scheduler.go:456]: have some duplicate with line435

File: br/pkg/lightning/backend/external/writer.go [hunk 3]
```diff
@@ -168,6 +170,13 @@ func (b *WriterBuilder) SetBlockSize(blockSize int) *WriterBuilder {
 	return b
 }
 
+// SetGroupOffset set the group offset of a writer.
+// This can be used to group the summaries from different writers.
+func (b *WriterBuilder) SetGroupOffset(offset int) *WriterBuilder {
```
Comment (wjhuang2016) [inline br/pkg/lightning/backend/external/writer.go:175]: We can comment it is used for `multi-schema-change`

**Review by wjhuang2016 — APPROVED** [review]

**Comment by ywqzzy** [comment]
Will it affect backward compatibility？

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**